tests/acceptance/30_generic_methods/permissions_group_acl_present.cf

```diff
@@ -0,0 +1,738 @@
+
+     # SUCCESS
+      "files[13]"     string => "file13";
+      "recursive[13]" string => "true";
+      "group[13]"      string => "bin";
+      "ace[13]"       string => "+rwx";
+      "create[13]"    string => "true";
+      "initial[13]"    slist => { "user:*:rwx", "group:*:rx", "group:bin:rwx", "user:bin:r", "all:r" };
+      "exp_lines[13]"  slist => { "user::rwx", "group::r-x", "group:bin:rwx", "user:bin:r--", "other::r--" };
+      "mode[13]"      string => "enforce";
+      "status[13]"    string => "success";
+      "execute[13]"   string => "true";
+
+
+      "files[14]"     string => "file14";
+      "recursive[14]" string => "true";
+      "group[14]"      string => "bin";
+      "ace[14]"       string => "-rw";
+      "create[14]"    string => "true";
+      "initial[14]"    slist => { "user:*:rwx", "group:*:rx", "group:bin:x", "user:bin:r", "all:r" };
+      "exp_lines[14]"  slist => { "user::rwx", "group::r-x", "group:bin:--x", "user:bin:r--", "other::r--" };
+      "mode[14]"      string => "enforce";
+      "status[14]"    string => "success";
+      "execute[14]"   string => "true";
+
+      "files[15]"     string => "file15/.";
+      "recursive[15]" string => "true";
+      "group[15]"      string => "bin";
+      "ace[15]"       string => "-rw";
+      "create[15]"    string => "true";
+      "initial[15]"    slist => { "user:*:rwx", "group:*:rx", "group:bin:-rwx", "user:bin:r", "all:rx" };
+      "exp_lines[15]"  slist => { "user::rwx", "group::r-x", "group:bin:---", "user:bin:r", "other::r-x" };
+      "mode[15]"      string => "enforce";
+      "status[15]"    string => "success";
+      "execute[15]"   string => "true";
+
+      "files[16]"     string => "file15/subfile1";
+      "create[16]"    string => "true";
+      "initial[16]"    slist => { "user:*:rwx", "group:*:rx", "group:bin:x", "user:bin:r", "all:r" };
+      "exp_lines[16]"  slist => { "user::rwx", "group::r-x", "group:bin:--x", "user:bin:r--", "other::r--" };
+
+      "files[17]"     string => "file17/.";
+      "recursive[17]" string => "true";
+      "group[17]"      string => "bin";
+      "ace[17]"       string => "-rw";
+      "create[17]"    string => "true";
+      "initial[17]"    slist => { "user:*:rwx", "group:*:rx", "group:bin:x", "user:bin:r", "all:rx" };
+      "exp_lines[17]"  slist => { "user::rwx", "group::r-x", "group:bin:--x", "user:bin:r--", "other::r-x" };
+      "mode[17]"      string => "enforce";
+      "status[17]"    string => "success";
+      "execute[17]"   string => "true";
+
+      # non-recursive
+      "files[18]"     string => "file18";
+      "recursive[18]" string => "false";
+      "group[18]"      string => "*";
+      "ace[18]"       string => "+rx";
+      "create[18]"    string => "true"; 
+      "initial[18]"    slist => { "user:*:rwx", "group:*:rx", "group:bin:r", "user:bin:r", "all:r" };
+      "exp_lines[18]"  slist => { "user::rwx", "group::r-x", "group:bin:r--", "user:bin:r--", "other::r--" };
```

Comment:
shouldn't group:bin: be r-x ?

Reply:
"*" is in cfengine reference the current owner, in this case it just change the global "group" section, which is in fact the classic unix right for group owner.